### ISSUE TITLE
[service] Unpredictable init_once_begin_initialize link errors in  Visual Studio 16 agent

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan-center-index#4826]
Visual Studio 16 agent randomly fails to link symbols like init_once_begin_initialize (should live in kernel32: https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-initoncebegininitialize). We have to close/open PR until it works.

Affected PR:
https://github.com/conan-io/conan-center-index/pull/4810
https://github.com/conan-io/conan-center-index/pull/4798
https://github.com/conan-io/conan-center-index/pull/4752 (/cc @mathbunnyru)

**Comment by SpaceIm** [comment]
Yes, it's exactly the issue we have I think.

**Comment by anton-danielsson** [comment]
Yeah, there was an ABI break in 16.5, see: https://docs.microsoft.com/en-us/cpp/overview/cpp-conformance-improvements?view=msvc-160#tls-guard-changes and since Conan only considers major MSVC versions this becomes a bit fragile.

There is a longer discussion that touches upon this here: https://github.com/conan-io/conan/issues/3311
TLDR: It will be fixed by the new `msvc` compiler in settings.yml.

**Comment by bredej** [comment]
Here's a [workaround](https://developercommunity.visualstudio.com/t/-imp-std-init-once-complete-unresolved-external-sy/1684365#T-N10041864) and [here's](https://developercommunity.visualstudio.com/t/-imp-std-init-once-complete-unresolved-external-sy/1684365#T-N10047968) some background information.

The TLDR
```
#if _MSC_VER >= 1932 // Visual Studio 2022 version 17.2+
#    pragma comment(linker, "/alternatename:__imp___std_init_once_complete=__imp_InitOnceComplete")
#    pragma comment(linker, "/alternatename:__imp___std_init_once_begin_initialize=__imp_InitOnceBeginInitialize")
#endif
```

**Comment by jgsogo** [comment]
Hi! I think the best thing we can do is to ensure that all windows servers are using exactly the same VS version (first be deterministic, then the rest). Now I can see that two versions are reported for VS 2017 (and VS 2019): 

```
OK  : Windows | ansiblewindows6 | Visual Studio 15/2017/19.1x - Found 19.16.27048
OK  : Windows | ansiblewindows6 | Visual Studio 16/2019/19.2x - Found 19.29.30143
```
vs
```
OK  : Windows | ansiblewindows4 | Visual Studio 15/2017/19.1x - Found 19.16.27045
OK  : Windows | ansiblewindows4 | Visual Studio 16/2019/19.2x - Found 19.29.30141
```

The only sustainable policy can be to _"install latest update"_ and I think it is what we should commit to. Whenever we provision a new Windows server we need to either update all of them or provision all instances and decommission previous ones.

Of course there is still an underlying issue, if that last bit is not included in the packageID (given we are using compiler `Visual Studio`, not `msvc`) we might have binaries that will fail to link 😓 , but we can only build them again.

Does it sound good?

**Comment by mathbunnyru** [comment]
@jgsogo sounds good to me.
I think it's the best solution at the moment.

**Comment by jgsogo** [comment]
It's been working like this for a while now. Whenever we deploy a new server... we recreate all of them.
